pkgs/by-name/fe/fedimint/package.nix

```diff
@@ -0,0 +1,98 @@
+{ lib
+, fetchFromGitHub
+, openssl
+, pkg-config
+, protobuf
+, rustPlatform
+, stdenv
+, libiconv
+, clang
+, libclang
+, Security
+, SystemConfiguration
+}:
+let
+  gitRev = "bcc85e8201615b464f9369371d6640823ff12a1a";
+in
+rustPlatform.buildRustPackage {
+  pname = "fedimint";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "fedimint";
+    repo = "fedimint";
+    rev = gitRev;
+    hash = "sha256-eOGmx/freDQLxZ48nP9Y2kA84F6sdig6qfZwnJfOB3g=";
+  };
+
+  cargoHash = "sha256-qK9IvsVLoppwa+BkOi2RqrBY3V97tXS8tZz5Rq7ilH8=";
+
+  nativeBuildInputs = [
+    protobuf
+    pkg-config
+    clang
+    libclang.lib
+  ];
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [
+    Security
+    libiconv
+    Security
+    SystemConfiguration
+  ];
+
+  outputs = [ "out" "fedimintCli" "fedimint" "gateway" "gatewayCli" "devimint" ];
```

Comment:
Isn't that overkill and just making things complicated?

Reply:
I'm not sure. These are relatively heavy (because Rust, and because these binaries include a logic-heavy Fedimint client library), and Fedimint is a distributed system with multiple actors with different roles. There are clients, there are distributed peers in a federation, there are gateways, there are binaries integrating gateways with other systems as plugins, there is a sandbox testing utility, etc.

Once I settled on multiple-outputs approach, it seemed relatively simple to just throw everything in into role-specific output. IIUC, whoever wants all binaries together can just use the default output.